test/integration/stats_integration_test.cc

```diff
@@ -25,6 +29,35 @@ class StatsIntegrationTest : public testing::TestWithParam<Network::Address::IpV
   void initialize() override { BaseIntegrationTest::initialize(); }
 };
 
+class ClusterMemoryUtilization : public TestBaseWithParam<Network::Address::IpVersion>,
+                                 public BaseIntegrationTest {
+public:
+  ClusterMemoryUtilization() : BaseIntegrationTest(GetParam(), realTime()) {}
+
+  void TearDown() override {
+    test_server_.reset();
```

Comment:
Is this override needed? Won't normal destruction suffice here?

Reply:
Sorry I didn't realize this wasn't done yet; this should definitely be cleaned up before approval, also htuch's comment below.